go-controller/pkg/ovn/pods.go

```diff
@@ -414,6 +429,14 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			pod.Namespace, pod.Name, time.Since(start), libovsdbExecuteTime, podAnnoTime)
 	}()
 
+	// it is possible to try to add a pod here that has no node. For example if a pod was deleted with
+	// a finalizer, and then the node was removed. In this case the pod will still exist in a running state.
+	if _, err := oc.watchFactory.GetNode(pod.Spec.NodeName); kerrors.IsNotFound(err) &&
+		util.PodTerminatingWithFinalizer(pod) {
```

Comment:
I debated here whether or not to even just skip addLogicalPort if a pod is terminating with finalizer. In the end I felt like there could be some unknown case in the future that I could be missing so I decided to leave it alone. Once the pod is deleted we would get the delete event later and clean it up anyway.

Reply:
good question, I know this specific fix is tacking the what if the node is gone case, but we probably need to do the same for all cases where pod finalizers are set? (thinking if that is going to have unforseen consequences... which is why we wanna do case by case?)

Reply:
Because if the pod terminating with finalizer, it is still technically running so we should ensure it still exists. Here is the case I was thinking of:
1. ovnkube-master restarts, lets assume OVN NBDB was deleted
2. sync pods happens. We see this pod exists and we allocate the IP
3. watchPods() now adds all existing pods. Here we encounter this pod. We should ensure it still has a logical_port in OVN I suppose.

Like I said in my first comment I wasn't sure what to do here. Because if the pod is terminating with a finalizer then the containers have exited, so you would think its OK to ignore wiring its networking. However there could be some unknown case where it matters so I decided to just leave it alone (and then a subsequent delete event later will clean it up).

The Node is relevant because we could get an add here without the node existing, and in that case we simply cannot wire this pod.